src/roswire/ros2/reader.py

```diff
@@ -0,0 +1,141 @@
+        # TODO Implement launch config
+        lc = LaunchConfig()
+        ctx = LaunchContext(namespace='/', filename=filename)
+        if argv:
+            ctx = ctx.with_argv(argv)
+
+        ctx, cfg = self._load_launch_objects(ctx, lc,
+                                             [list(config_nodes)])
+        logger.debug(f'launch configuration: {cfg}')
+        return lc
+
+    def _load_launch_objects(self, ctx: LaunchContext,
+                             cfg: LaunchConfig,
+                             node_list: Sequence[Sequence[Dict]]
+                             ) -> Tuple[LaunchContext, LaunchConfig]:
+        for nodes in node_list:
+            for node in nodes:
+                if node['__TYPE__'] == 'Node':
+                    print(f'processing: {node}')
+                    # remappings = self.\
+                    #     _get_remappings_from_json(node.get('remappings'))
+                    args: Optional[Any] = node.get('args', [])
+                    if isinstance(args, list):
+                        args = ' '.join(args)
+                    print(args)
+                    remappings = node.get('remappings', [])
+                    if remappings is not None:
+                        remappings = tuple(remappings)
+                    nc = NodeConfig(
+                        name=node['name'],
+                        namespace=node['namespace'],
+                        package=node['package'],
+                        executable_path=node['executable_path'],
+                        executable_type=ExecutableType[
+                            node['executable_type']
+                        ],
+                        remappings=remappings, # noqa
+                        filename=node.get('filename'),
+                        output=node.get('output'),
+                        required=self._get_bool(node.get('required'), False),
+                        respawn=self._get_bool(node.get('respawn'), False),
+                        respawn_delay=self._get_float(
+                            node.get('respawn_delay'), 0.0),
+                        env_args=tuple(node.get('env_args', [])), # noqa
+                        cwd=node.get('cwd'),
+                        args=args,
+                        launch_prefix=node.get('launch_prefix'),
+                        typ=''
+                    )
+                    cfg.with_node(nc)
+                elif node['__TYPE__'] == 'ExecuteProcess' \
+                        and node['cmd'][0] == 'gazebo':
+                    nc = NodeConfig(
+                        name='gazebo',
+                        namespace='',
+                        typ='',
+                        package='',
+                        executable_path=node['cmd'][0],
+                        executable_type=ExecutableType.LIKELY_CPP,
+                        args=' '.join(node['cmd'][1:]),
+                    )
+                    cfg.with_node(nc)
```

Comment:
Since `cfg` is immutable, `with_node(...)` is a side-effect free method that returns a new config. Should update this (and other cases) to `cfg = cfg.with_node(nc)`.

Reply:
Done